addons/sale_timesheet/controllers/portal.py

```diff
@@ -63,6 +64,26 @@ def _get_groupby_mapping(self):
             invoice='timesheet_invoice_id')
         return groupby_mapping
 
+    def _task_get_page_view_values(self, task, access_token, **kwargs):
+        values = super()._task_get_page_view_values(task, access_token, **kwargs)
+        try:
+            if values['task'].sale_order_id and self._document_check_access('sale.order', values['task'].sale_order_id.id):
```

Comment:
`values['task']` seem to be the exact same object as `task`, why bother accessing it through `values` every time?